### DISCUSSION TITLE
coordinates to voxel index

### DISCUSSION THREAD

**Discussion by wensoffthewall** [e0404/matRad#563] [Q&A] [answered]
MatRad is absolutely wonderful, I'd like to use it for research purposes. I've been struggling  with the coordinates of the proton spots. Is it possible to somehow get the CT coordinates of the spots?

**Comment by wahln** [comment, marked as answer]
This is indeed not that straightforward or natively implemented. matRad uses the geometrical coordinates to define "rays" (you can get their coordinate w.r.t to the isocenter in world coordinates by `stf(iBeam).ray(iRay).rayPos`.
Spot positions are then chosen on each ray based on the WEPL computed from voxel ray-tracing such that the target is covered, and the chosen energies are stored in each ray.

To get the coordinate of a peak then, one would need to use the raytracer to extract the WEPL in each voxel passed by the ray and then  seek/interpolate the peakpositions from the basedata-file for all the given energies. Maybe this can be hacked into the matRad_generateStf function, such that the geometrical world coordinates / coordinates in BEV of the spots are stored as well.
